# Apostoli Karpouzis Final Project Notebook

# Submetric: Least Crime 
    In this notebook, I compare populations of each Pittsburgh neighborhood to their reported crimes in the last 30 days. This will produce a crime rate per capita for each neighborhood. The lowest rate per capita will give us an idea of the safest neighborhood to live in in Pittsburgh. 

**First, I import pandas, as well as the csv files I will be using as my submetric.**

In [ ]:
import pandas as pd
crime = pd.read_csv('crime.csv')
pop = pd.read_csv('pops.csv')

**I started by printing out the neighborhoods in order of most reported crimes, to least.**

In [3]:
grouped = crime.groupby(crime.columns[6])
counts = grouped.size().sort_values(ascending=False)
for value, count in counts.items():
    print(f'{value}: {count}')

Central Business District: 368
South Side Flats: 127
Carrick: 90
East Liberty: 83
Marshall-Shadeland: 74
Mount Washington: 74
Bluff: 74
Shadyside: 68
Crawford-Roberts: 66
Sheraden: 65
Brookline: 63
Homewood North: 58
Lincoln-Lemington-Belmar: 58
Bloomfield: 57
Homewood South: 56
Squirrel Hill South: 55
Brighton Heights: 53
East Allegheny: 48
Hazelwood: 47
Beechview: 46
Perry South: 44
South Side Slopes: 42
Strip District: 39
Bedford Dwellings: 38
Larimer: 38
North Oakland: 36
Knoxville: 35
Elliott: 34
Chateau: 33
Allentown: 33
Middle Hill: 33
East Hills: 32
Highland Park: 31
Central Lawrenceville: 30
Crafton Heights: 30
Garfield: 30
Central North Side: 29
North Shore: 25
Spring Hill-City View: 24
Perry North: 23
Central Oakland: 23
California-Kirkbride: 21
Beltzhoover: 21
Point Breeze North: 21
Northview Heights: 20
Greenfield: 20
Homewood West: 20
Point Breeze: 20
Squirrel Hill North: 20
Lower Lawrenceville: 20
Allegheny Center: 19
Upper Lawrenceville: 18
Arlington: 18
Banksville: 16


**After seeing the Central Business District (Downtown) first on the list, I realized crime alone wasn't enough to categorize a neighborhood as dangerous. 
Thats because Downtown obviously has a higher population than many of the other neighborhoods. This gave me the idea to bring in the 2020 concencus populations
to calculate a crime rate per capita. I print out the concensus populations next in alphabetical order.**

In [21]:
pd.set_option('display.max_rows', None)
pops = pop.loc[:, ['Neighborhood', '2020_Total_Population']]
print(pops)

                                Neighborhood  2020_Total_Population
0                           Allegheny Center                   1419
1                             Allegheny West                    540
2                                  Allentown                   2305
3   Arlington - Arlington Heights (Combined)                   2040
4                                 Banksville                   4127
5                          Bedford Dwellings                   1386
6                                  Beechview                   7596
7                                Beltzhoover                   1604
8                                 Bloomfield                   8665
9                                      Bluff                   5882
10                                   Bon Air                    759
11                          Brighton Heights                   7105
12                                 Brookline                  13063
13                      California-Kirkbride    

**Next, I reordered the crime reports in alphabetical order by neighborhood to line in up with the population data.**

In [5]:
grouped = crime.groupby(crime.columns[6])
counts = grouped.size()
for value, count in counts.items():
    print(f'{value}: {count}')

Allegheny Center: 19
Allegheny West: 6
Allentown: 33
Arlington: 18
Arlington Heights: 3
Banksville: 16
Bedford Dwellings: 38
Beechview: 46
Beltzhoover: 21
Bloomfield: 57
Bluff: 74
Bon Air: 12
Brighton Heights: 53
Brookline: 63
California-Kirkbride: 21
Carrick: 90
Central Business District: 368
Central Lawrenceville: 30
Central North Side: 29
Central Oakland: 23
Chartiers City: 1
Chateau: 33
Crafton Heights: 30
Crawford-Roberts: 66
Duquesne Heights: 16
East Allegheny: 48
East Carnegie: 1
East Hills: 32
East Liberty: 83
Elliott: 34
Esplen: 2
Fairywood: 10
Fineview: 15
Friendship: 15
Garfield: 30
Glen Hazel: 6
Greenfield: 20
Hays: 4
Hazelwood: 47
Highland Park: 31
Homewood North: 58
Homewood South: 56
Homewood West: 20
Knoxville: 35
Larimer: 38
Lincoln Place: 12
Lincoln-Lemington-Belmar: 58
Lower Lawrenceville: 20
Manchester: 15
Marshall-Shadeland: 74
Middle Hill: 33
Morningside: 10
Mount Oliver: 5
Mount Washington: 74
New Homestead: 3
North Oakland: 36
North Shore: 25
Northview Heights: 

**The tedious part: I wanted to divide the populations by the crime, but to do so I had to make sure all the neighborhoods were present in each and lined up, which after inspection they were close but not identical. I compared the two lists and took some notes on what needed changed before computing information between the two sets:**
- delete Outside City
- delete Outside County
- delete Outside State
- combine Arlington and Arlignton Heights
- change Spring Hill spelling (remove View)
- flip mt washington and mt oliver


I made the changes and the two lists now matched up: 

Allegheny Center: 19 Allegheny West: 6 Allentown: 33 Arlington - Arlington Heights (Combined): 21 Banksville: 16 Bedford Dwellings: 38 Beechview: 46 Beltzhoover: 21 Bloomfield: 57 Bluff: 74 Bon Air: 12 Brighton Heights: 53 Brookline: 63 California-Kirkbride: 21 Carrick: 90 Central Business District: 368 Central Lawrenceville: 30 Central North Side: 29 Central Oakland: 23 Chartiers City: 1 Chateau: 33 Crafton Heights: 30 Crawford-Roberts: 66 Duquesne Heights: 16 East Allegheny: 48 East Carnegie: 1 East Hills: 32 East Liberty: 83 Elliott: 34 Esplen: 2 Fairywood: 10 Fineview: 15 Friendship: 15 Garfield: 30 Glen Hazel: 6 Greenfield: 20 Hays: 4 Hazelwood: 47 Highland Park: 31 Homewood North: 58 Homewood South: 56 Homewood West: 20 Knoxville: 35 Larimer: 38 Lincoln Place: 12 Lincoln-Lemington-Belmar: 58 Lower Lawrenceville: 20 Manchester: 15 Marshall-Shadeland: 74 Middle Hill: 33 Morningside: 10 Mount Washington: 74 Mt. Oliver: 5 New Homestead: 3 North Oakland: 36 North Shore: 25 Northview Heights: 20 Oakwood: 10 Overbrook: 12 Perry North: 23 Perry South: 44 Point Breeze: 20 Point Breeze North: 21 Polish Hill: 5 Regent Square: 2 Ridgemont: 1 Shadyside: 68 Sheraden: 65 South Oakland: 15 South Shore: 9 South Side Flats: 127 South Side Slopes: 42 Spring Garden: 7 Spring Hill-City: 24 Squirrel Hill North: 20 Squirrel Hill South: 55 St. Clair: 5 Stanton Heights: 15 Strip District: 39 Summer Hill: 4 Swisshelm Park: 4 Terrace Village: 16 Troy Hill: 12 Upper Hill: 16 Upper Lawrenceville: 18 West End: 7 West Oakland: 12 Westwood: 15 Windgap: 10

**I then imported re module so I could use the findall method to strip all the numbers away into a separate list**

In [6]:
import re

data = "Allegheny Center: 19 Allegheny West: 6 Allentown: 33 Arlington - Arlington Heights (Combined): 21 Banksville: 16 Bedford Dwellings: 38 Beechview: 46 Beltzhoover: 21 Bloomfield: 57 Bluff: 74 Bon Air: 12 Brighton Heights: 53 Brookline: 63 California-Kirkbride: 21 Carrick: 90 Central Business District: 368 Central Lawrenceville: 30 Central North Side: 29 Central Oakland: 23 Chartiers City: 1 Chateau: 33 Crafton Heights: 30 Crawford-Roberts: 66 Duquesne Heights: 16 East Allegheny: 48 East Carnegie: 1 East Hills: 32 East Liberty: 83 Elliott: 34 Esplen: 2 Fairywood: 10 Fineview: 15 Friendship: 15 Garfield: 30 Glen Hazel: 6 Greenfield: 20 Hays: 4 Hazelwood: 47 Highland Park: 31 Homewood North: 58 Homewood South: 56 Homewood West: 20 Knoxville: 35 Larimer: 38 Lincoln Place: 12 Lincoln-Lemington-Belmar: 58 Lower Lawrenceville: 20 Manchester: 15 Marshall-Shadeland: 74 Middle Hill: 33 Morningside: 10 Mount Washington: 74 Mt. Oliver: 5 New Homestead: 3 North Oakland: 36 North Shore: 25 Northview Heights: 20 Oakwood: 10 Overbrook: 12 Perry North: 23 Perry South: 44 Point Breeze: 20 Point Breeze North: 21 Polish Hill: 5 Regent Square: 2 Ridgemont: 1 Shadyside: 68 Sheraden: 65 South Oakland: 15 South Shore: 9 South Side Flats: 127 South Side Slopes: 42 Spring Garden: 7 Spring Hill-City: 24 Squirrel Hill North: 20 Squirrel Hill South: 55 St. Clair: 5 Stanton Heights: 15 Strip District: 39 Summer Hill: 4 Swisshelm Park: 4 Terrace Village: 16 Troy Hill: 12 Upper Hill: 16 Upper Lawrenceville: 18 West End: 7 West Oakland: 12 Westwood: 15 Windgap: 10"
numbers = re.findall('\d+', data)
print(numbers)

['19', '6', '33', '21', '16', '38', '46', '21', '57', '74', '12', '53', '63', '21', '90', '368', '30', '29', '23', '1', '33', '30', '66', '16', '48', '1', '32', '83', '34', '2', '10', '15', '15', '30', '6', '20', '4', '47', '31', '58', '56', '20', '35', '38', '12', '58', '20', '15', '74', '33', '10', '74', '5', '3', '36', '25', '20', '10', '12', '23', '44', '20', '21', '5', '2', '1', '68', '65', '15', '9', '127', '42', '7', '24', '20', '55', '5', '15', '39', '4', '4', '16', '12', '16', '18', '7', '12', '15', '10']


**Have to change the string numbers to ints.**

In [22]:
numbers = ['19', '6', '33', '21', '16', '38', '46', '21', '57', '74', '12', '53', '63', '21', '90', '368', '30', '29', '23', '1', '33', '30', '66', '16', '48', '1', '32', '83', '34', '2', '10', '15', '15', '30', '6', '20', '4', '47', '31', '58', '56', '20', '35', '38', '12', '58', '20', '15', '74', '33', '10', '74', '5', '3', '36', '25', '20', '10', '12', '23', '44', '20', '21', '5', '2', '1', '68', '65', '15', '9', '127', '42', '7', '24', '20', '55', '5', '15', '39', '4', '4', '16', '12', '16', '18', '7', '12', '15', '10']
numbers = list(map(int, numbers))
print(numbers)

[19, 6, 33, 21, 16, 38, 46, 21, 57, 74, 12, 53, 63, 21, 90, 368, 30, 29, 23, 1, 33, 30, 66, 16, 48, 1, 32, 83, 34, 2, 10, 15, 15, 30, 6, 20, 4, 47, 31, 58, 56, 20, 35, 38, 12, 58, 20, 15, 74, 33, 10, 74, 5, 3, 36, 25, 20, 10, 12, 23, 44, 20, 21, 5, 2, 1, 68, 65, 15, 9, 127, 42, 7, 24, 20, 55, 5, 15, 39, 4, 4, 16, 12, 16, 18, 7, 12, 15, 10]


**Now that we have the Population data in a csv and a list of the numbers we want to divide that column by, we can calculate the crime rate per capita in Pitssburgh neighborhoods:**

In [18]:
pop['2020_Total_Population'] = pop['2020_Total_Population'].astype(int)
perCapita = pop['2020_Total_Population'].div(pd.Series(numbers)).astype(int)
print(perCapita)


0      74
1      90
2      69
3      97
4     257
5      36
6     165
7      76
8     152
9      79
10     63
11    134
12    207
13     33
14    114
15     14
16    157
17    102
18    253
19    492
20      0
21    133
22     34
23    149
24     39
25    491
26     90
27     74
28     64
29    161
30    109
31     76
32    130
33    121
34    109
35    363
36     80
37     82
38    201
39     45
40     40
41     32
42     99
43     40
44    262
45     62
46    132
47    135
48     54
49     53
50    325
51    117
52     88
53    305
54    296
55     12
56     88
57    102
58    290
59    163
60     85
61    267
62     94
63    240
64    485
65    390
66    225
67     79
68    221
69      3
70     58
71    103
72    112
73     98
74    593
75    278
76     36
77    295
78     47
79    249
80    334
81    145
82    220
83    107
84    133
85     29
86    211
87    222
88    134
dtype: int64


**I figured the above numbers were a little confusing, so I tried to fix the format in the following code. The numbers above represent 1 crime per that many people in each neighborhood.**

In [24]:
print("In the last 30 days, crime per capita for each Pittsburgh neighborhood is as shown.")
print(pop['Neighborhood'] + ": " + perCapita.astype(str) + " people per crime.")

In the last 30 days, crime per capita for each Pittsburgh neighborhood is as shown.
0                Allegheny Center: 74 people per crime.
1                  Allegheny West: 90 people per crime.
2                       Allentown: 69 people per crime.
3     Arlington - Arlington Heights (Combined): 97 p...
4                     Banksville: 257 people per crime.
5               Bedford Dwellings: 36 people per crime.
6                      Beechview: 165 people per crime.
7                     Beltzhoover: 76 people per crime.
8                     Bloomfield: 152 people per crime.
9                           Bluff: 79 people per crime.
10                        Bon Air: 63 people per crime.
11              Brighton Heights: 134 people per crime.
12                     Brookline: 207 people per crime.
13           California-Kirkbride: 33 people per crime.
14                       Carrick: 114 people per crime.
15    Central Business District (Downtown): 14 peopl...
16         Central L

# The Best Neighborhood: Squirell Hill

# Conclusion:

    I want to begin with disclaimers/flaws in this method. The main thing that is overlooked in my analysis of crime is the type of crime reported. Because looking at crime is essentially the method to find out the safest neighborhood, crimes such as "public drunkness' would not necessarily make a neighborhood less safe. However, I did a quick scan of the crimes reported and the cases that we would want to ignore when considering saftey appeared negligible in numbers compared to the dangerous and harfmful crime. The other flaw would simply be that this is an estimate based off of a population reported 3 years ago and some minor rounding in calculations, but I still think it holds mostly true a few years later. 

**Anyways...**

    After analyzing the crime rates per captia, you can see Squirell Hill North has the lowest rate at 1 report for every 593 people. I think it is necessary to consider population when looking at crime rates becuase crime rates alone dont give you the full picture. For instance, in the last 30 days, South Shore reported 9 instances of crime which is less than half of Squirell Hill's 20. However, South Shore only reported 29 people on the 2020 concencus, resulting in a crime per every 3 people. Thus, even though Squirell Hill reported more crime as a neighborhood, you are far less likely to come close to it if you were to live here due to a much larger population presence in the neighborhood. This makes Squirrel Hill the best neighborhood to live in in accordance with safety.